share/spice/search_code/search_code.js

```diff
@@ -1,52 +1,67 @@
-function ddg_spice_search_code(api_response) {
+(function(env) {
     "use strict";
+    env.ddg_spice_search_code = function(api_result) {
 
-    var query = api_response.query;
-    var data = api_response.results;
+        var query = api_result.query;
+        var data = api_result.results;
 
-    if(!data.length || data.length === 0) {
-        return Spice.failed('search_code');
-    }
-
-    var result;
-    for (var i = 0; i < data.length; i++) {
-        var checkRelevancy = [data[i].name, data[i].displayname, data[i].namespace].join(" ");
-        if (DDG.isRelevant(checkRelevancy, [], 2)) {
-            result = data[i];
-            break;
+        if(!data.length || data.length === 0) {
+            return Spice.failed('search_code');
         }
-    }
 
-    if (!result) {
-        return Spice.failed('search_code');
-    }
+        var result;
+        for (var i = 0; i < data.length; i++) {
+            var checkRelevancy = [data[i].name, data[i].displayname, data[i].namespace].join(" ");
+            if (DDG.isRelevant(checkRelevancy, [], 2)) {
+                result = data[i];
+                break;
+            }
+        }
 
-    function formatName(result) {
-        var formatted_name = result.name;
+        if (!result) {
+            return Spice.failed('search_code');
+        }
 
-        if (result.displayname !== '' || result.namespace !== '') {
-            formatted_name += ' (';
-            if (result.displayname !== '') {
-                formatted_name += result.displayname;
+        function formatTitle(result) {
+            var formatted_title = result.name;
+            if (result.displayname && result.displayname !== '') {
+                formatted_title += ' (' + result.displayname + ')';
             }
-            if (result.namespace !== '') {
-                formatted_name += (result.displayname ? ', ' : '') + result.namespace;
+            return formatted_title;
+        }
+
+        function formatSubtitle(result) {
+            var formatted_subtitle = null;
+            if (result.namespace && result.namespace !== '') {
+                formatted_subtitle = result.namespace;
             }
-            formatted_name += ')';
+            return formatted_subtitle;
         }
 
-        return formatted_name;
-    }
+        result.title = formatTitle(result);
+        result.subtitle = formatSubtitle(result);
 
-    Spice.add({
-        data             : result,
-        header1          : formatName(result),
-        sourceUrl       : 'https://searchcode.com/?q=' + query,
-        sourceName      : 'search[code]',
-        templates: {
-            item: Spice.search_code.search_code,
-            detail: Spice.search_code.search_code
-        },
-        
-    });
-}
+        Spice.add({
+            id: 'search_code',
+            name: 'Answer',
+            data: result,
+            meta: {
+                sourceURL: 'https://searchcode.com/?q=' + query,
```

Comment:
Putting the reference link in `subtitle` isn't a good idea here. Because `subtitle` won't be displayed in every case. Is there any way around this? @moollaza